packages/common/src/types/common.ts

```diff
@@ -23,3 +23,68 @@ export type AnyFunction = (...args: any) => any;
 export type ReplaceProperties<T extends AnyObject, R extends AnyObject> = {
   [K in keyof T]: K extends keyof R ? R[K] : T[K];
 } & AnyObject;
+
+/**
+ * Never allow any properties of `Type`.
+ *
+ * Utility type, probably never a reason to export.
+ */
+type Never<Type> = {
+  [Property in keyof Type]?: never;
```

Comment:
Nit: when referring to key types we can simply use `K`.

```suggestion
  [K in keyof Type]?: never;
```